crates/stand-alone-test/src/bin/search_hnsw.rs

```diff
@@ -0,0 +1,84 @@
+fn main() {
+    #[cfg(feature = "stand-alone-test")]
+    search_hnsw();
+}
+
+#[cfg(feature = "stand-alone-test")]
+fn search_hnsw() {
+    use base::{
+        index::{IndexOptions, OptimizingOptions, SearchOptions, SegmentsOptions, VectorOptions},
+        scalar::F32,
+        search::{Filter, Payload},
+        vector::Vecf32Borrowed,
+    };
+    use stand_alone_test::utils::read_vecs_file;
+    use std::path::Path;
+
+    #[derive(Clone)]
+    struct FilterStruct {}
+    impl Filter for FilterStruct {
+        fn check(&mut self, _: Payload) -> bool {
+            true
+        }
+    }
+
+    let dims = 128;
+    let path = Path::new("/home/yanqi/stand-alone-test/data/hnsw");
```

Comment:
We could use `clap` to get path of datasets from users. Otherwise anyone who wants to run these tests needs to create a user named yanqi on their machines.